### DISCUSSION TITLE
Asyncify example?

### DISCUSSION THREAD

**Discussion by tareksander** [WebAssembly/binaryen#6399] [Q&A]
I haven't found an example for asyncify except the documentation, is this the correct order of things?

1. Call an exported WASM function
2. WASM calls some import that wants to unwind.
3. Set up asyncify stack and call `asyncify_start_unwind`
4. Return to WASM
5. Unwinding ends up at the place where WASM was first called, and `asyncify_stop_unwind` can be called there
6. When execution should continue, call `asyncify_start_rewind` and call the exported function again
7. This will end up in the same imported function again. Check if we're rewinding and call `asyncify_stop_rewind` and return to WASM
8. WASM execution now resumes where it left off

Is my mental model correct?

**Comment by kripken** [comment]
Yes, that sounds right.

The details here may be helpful:

https://kripken.github.io/blog/wasm/2019/07/16/asyncify.html

You can also walk through the modified code from a simple example to get a feel for things.
